lib/utils/UserManager.dart

```diff
@@ -204,9 +204,9 @@ class UserManager {
     return prefs.getStringList('imagesList');
   }
 
-  static Future<String?> getHashtag() async {
+  static Future<List<String>?> getHashtags() async {
     var prefs = await SharedPreferences.getInstance();
-    return prefs.getString('hashtag');
+    return prefs.getStringList('hashtags');
```

Comment:
It occurred to me that users who have already enabled the auto-tagging function will lose their tags due to the renaming of 'hashtag' to 'hashtags'.

To prevent this, we should consider applying a migration during app initialization. I was thinking on something like:
```dart
Future<void> migrateHashtagToHashtags() async {
  final prefs = await SharedPreferences.getInstance();

  // Check if the old variable exists
  if (prefs.containsKey('hashtag')) {
    // Read the old variable's value
    String? oldHashtag = prefs.getString('hashtag');

    // Save the value to the new variable as a list
    if (oldHashtag != null) {
      await prefs.setStringList('hashtags', [oldHashtag]);
    }

    // Remove the old variable
    await prefs.remove('hashtag');
  }
}
```

Reply:
Ei, Santi just told me that the users are currently setting the tags by specifying '#' as prefix. We need to take that into account for the migration... May be applying a extraction function in the function in the previous comment (using regex):
```dart
// Function to extract hashtags
  final RegExp hashtagRegExp = RegExp(r'#(\w+)');
  return hashtagRegExp
      .allMatches(input)
      .map((match) => match.group(1)!)
      .toList();
}
```

or using split(' ').startswith('#') ....

Reply:
I've added this and wrote a test with the 2 cases to check it's migrated as expected
![image](https://github.com/user-attachments/assets/2fd92e4e-8e57-405b-9001-26caf1c25614)